src/Shared/AppModels/AppDebug.cs

```diff
@@ -0,0 +1,229 @@
+                Content = "Are you sure you want to delete user data? If you hit yes, the app will restart.",
+                PrimaryButtonText = "Yes",
+                SecondaryButtonText = "No",
+                PrimaryButtonCommand = PerformUserDataDeletionCommand
+            };
+
+            await _deleteUserDataDialog.ShowAsync();
+        }
+
+        [RelayCommand]
+        private async Task PerformUserDataDeletionAsync()
+        {
+            try
+            {
+                var localDataFolder = ApplicationData.Current.LocalFolder.Path;
+                var localCache = ApplicationData.Current.LocalCacheFolder.Path;
+
+                var folder = await StorageFolder.GetFolderFromPathAsync(localDataFolder);
+                await folder.DeleteAsync();
+
+                folder = await StorageFolder.GetFolderFromPathAsync(localCache);
+                await folder.DeleteAsync();
+
+                var result = await CoreApplication.RequestRestartAsync("Application Restart Programmatically.");
+
+                if (result == AppRestartFailureReason.NotInForeground ||
+                   result == AppRestartFailureReason.RestartPending ||
+                   result == AppRestartFailureReason.Other)
+                {
+                    await ShowRetryContentDialogAsync($"Restart Failed. Please restart the app manually.", RestartAppCommand, null);
+                }
+            }
+            catch (Exception ex)
+            {
+                Guard.IsNotNull(_deleteUserDataDialog);
+
+                _deleteUserDataDialog.Hide();
+                await ShowRetryContentDialogAsync($"Couldn't delete the user data.", PerformUserDataDeletionCommand, ex);
+            }
+        }
+
+        private async Task ShowRetryContentDialogAsync(string error, ICommand retryCommand, Exception? ex)
+        {
+            Logger.LogError(error, ex);
+            var retryConfirmationDialog = new ContentDialog
+            {
+                Title = error,
+                Content = new StackPanel
+                {
+                    Width = 275,
+                    Spacing = 15,
+                    Children =
+                        {
+                            new Expander
+                            {
+                                Width = 275,
+                                Header = "View error",
+                                ExpandDirection = ExpandDirection.Down,
+                                HorizontalContentAlignment = HorizontalAlignment.Stretch,
+                                Content = new TextBlock { Text = $"Reason: {ex}", FontSize = 11, IsTextSelectionEnabled = true },
+                            },
+                        },
+                },
+                CloseButtonText = "Ignore, remove source",
+                PrimaryButtonText = "Try again",
+                PrimaryButtonCommand = retryCommand,
+            };
+
+            await retryConfirmationDialog.ShowAsync();
```

Comment:
Please merge feature/settings-ui into this branch, and add a ShowType enum in the first parameter. This new extension method allows us to queue or interrupt any ContentDialog that's already open, instead of the default behavior of just crashing.

Reply:
Upon testing: QueueNext didn't work as expected
[image](https://user-images.githubusercontent.com/15306909/209938519-ba4ac8c6-5d4e-45ce-a83c-d3fcb14a6852.png)
[image](https://user-images.githubusercontent.com/15306909/209938584-4f2dd046-70c4-4936-8476-35954f87d664.png)

ShowType Inturrupted works but crashes on retry by saying _inturrupted dialog is null
Parameter "_interruptedDialog" (Windows.UI.Xaml.Controls.ContentDialog) must be null, was <Windows.UI.Xaml.Controls.ContentDialog> (Windows.UI.Xaml.Controls.ContentDialog).
Parameter name: _interruptedDialog
in short both of them have issues
i'll undo my changes for now and let you fix these first